### ISSUE TITLE
The tree algorithm is not complete (Fixed)

### ISSUE THREAD

**Issue by Nikl174** [majorx234/haskell_examples#1]
Let's suppose a tree like this:
```
              0
            /   \
          1      10
        /   \      
       2     3      
            / \
           4   5
          / \ / \
         6  7 8  9
              .
              .
              .
```
Obviously the subtree with root in 3 is the biggest complete one.
But the algorithm **find_lagest_complete_subtree** just lets this subtree disappear.
In the round, where 1 is the current root, it calculates the size of the largest subtree in the branches (1 for root 2 and 7 for root 3, and this could be bigger though) then it takes the **minimum** (1) and completely forgets the other one.  
Then it go's back up the recursive tree to 0 ( so 2\*1+1 = 3 for root 1 and  1 for root 10) and now the biggest complete subtree is 2\*1+1 =3 big  but we already found a bigger one.

So I don't think, that we can use the minimum in this case.

**Comment by majorx234** [comment]
Ohh thx you are right,
have to think over again!
WIP

**Comment by majorx234** [comment]
Fixed 6d53ea79e2a14eda70a2b15d6a2a1a5bad8f68e7

**Comment by majorx234** [comment]
@Nikl174 Fixed 6d53ea79e2a14eda70a2b15d6a2a1a5bad8f68e7
